pkg/infra/log/file.go

```diff
@@ -39,61 +37,59 @@ type FileLogWriter struct {
 
 	Rotate    bool
 	startLock sync.Mutex
-}
-
-// an *os.File writer with locker.
-type MuxWriter struct {
+	logger    log.Logger
 	sync.Mutex
 	fd *os.File
 }
 
 // write to os.File.
-func (l *MuxWriter) Write(b []byte) (int, error) {
-	l.Lock()
-	defer l.Unlock()
-	return l.fd.Write(b)
+func (w *FileLogWriter) Write(b []byte) (int, error) {
+	w.docheck(len(b))
+	w.Lock()
+	defer w.Unlock()
+	return w.fd.Write(b)
 }
 
 // set os.File in writer.
-func (l *MuxWriter) setFD(fd *os.File) error {
-	if l.fd != nil {
-		if err := l.fd.Close(); err != nil && !errors.Is(err, os.ErrClosed) {
+func (w *FileLogWriter) setFD(fd *os.File) error {
+	if w.fd != nil {
+		if err := w.fd.Close(); err != nil && !errors.Is(err, os.ErrClosed) {
 			return fmt.Errorf("closing old file in MuxWriter failed: %w", err)
 		}
 	}
 
-	l.fd = fd
+	w.fd = fd
 	return nil
 }
 
 // create a FileLogWriter returning as LoggerInterface.
 func NewFileWriter() *FileLogWriter {
-	w := &FileLogWriter{
-		Filename: "",
-		Format:   log15.LogfmtFormat(),
-		Maxlines: 1000000,
-		Maxsize:  1 << 28, // 256 MB
-		Daily:    true,
-		Maxdays:  7,
-		Rotate:   true,
+	w := new(FileLogWriter)
```

Comment:
I prefer the syntax that was here before; what was wrong with it? :eyes:

Reply:
back to old syntax